File: Algorithms/GreedyAlgorithms/change/change.py
"""Least amount of coins
    change in terms of 1, 5, and 10 coins"""
import sys

def get_change(m):
    assert m > 1
    coins = 0
    while m > 0:
        if(m >= 1 and m < 5):
            m = m - 1
            coins += 1
        if(m >= 5 and m < 10):
            m = m - 5
            coins += 1
        if(m >= 10):
            m = m - 10
            coins += 1
    return coins 

if __name__ == "__main__":
    m = int(sys.argv[1])
    print(get_change(m), " Coins for ", m)

File: GreedyAlgorithms/change/change.cpp
#include<iostream>
#include<vector>
using namespace std;

int getChange(int n){
   vector<int> coins;
   while(n > 0){
       if(n >= 1 && n < 5){
           n -= 1;
           coins.push_back(1);
       }
       if(n >= 5 && n < 10){
           n -= 5;
           coins.push_back(5);
       }
       if(n >= 10){
           n -= 10;
           coins.push_back(10);
       }
   }
   for(int i = 0; i < coins.size(); i++){
       cout << coins[i] << " ";
   }
   return 0;
}

int main(){
    int n;
    cin >> n;
    cout << getChange(n) << endl;
    return 0;
}
